src/main/java/baseball/result/Result.java

```diff
@@ -0,0 +1,54 @@
+package baseball.result;
+
+import baseball.number.GameNumber;
+
+public class Result {
+    private int strikeCount;
+    private int ballCount;
+    public Result() {
+        this.strikeCount = 0;
+        this.ballCount = 0;
+    }
```

Comment:
저번 리뷰와 비슷한 부분인데 이곳에도 적용이 될 수 있을것 같아요.

`new Result()`를 보면 _모든 필드가 null로 초기화 되려나?_ 혹은 _아무런 필드도 안들고 있는 객체(무상태 객체)인가?_ 라는 생각이 들어요.

```suggestion
    public Result(int strikeCount, int ballCount) {
        this.strikeCount = strikeCount;
        this.ballCount = ballCount;
    }
```

처럼 바꾸어 `new Result(0, 0);`으로 사용하거나, 정적 팩토리 메서드를 통하여 초기 상태라는 것을 표현해줄 수 있지 않을까요?

_이 메서드를 통해 strikeCount와 ballCount가 0으로 초기화 될거야_ 라는 걸 사용하는 쪽에서 인지 할 수 있었으면 좋겠어요!

> 놀람 최소화 원칙(principle of least astonishment, principle of least surprise, POLA)은 사용자 인터페이스와 소프트웨어 설계에 적용되는 원칙이다. "필요한 기능에 크나큰 깜짝 놀래킬만한 요소가 있다면 해당 기능을 다시 설계할 필요가 있을 수 있다"는 것이 이 원칙의 일반적인 공식이다.
>
> 더 일반적으로 이야기하면 이 원칙은 시스템의 구성 요소가 대부분의 사용자들이 행동할 것으로 예측되는 방식으로 동작하는 것이 좋다는 것을 의미한다. 즉, 해당 동작이 사용자들을 놀래키지 않는 것이 좋다는 것이다.

Reply:
이 부분은 깊게 고민해 봐야겠어요 🤔